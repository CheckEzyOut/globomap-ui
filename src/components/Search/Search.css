.search {
  position: fixed;
  top: 0;
  left: 100px;
  bottom: 0;
  right: 0;
  display: flex;
  padding: 50px 7% 0;
  z-index: 8;
  background-color: white;
}
