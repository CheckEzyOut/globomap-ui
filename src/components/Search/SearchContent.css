/*
Copyright 2019 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.search-content,
.search-content-base,
.search-content-results,
.search-content-results th,
.search-content-results td,
.search-content-results tbody {
  box-sizing: border-box;
}

.search-content {
  position: fixed;
  left: 256px;
  bottom: 0;
  right: 0;
  padding: 20px 7% 0;
  background-color: #fff;
  overflow-y: auto;
  top: 0;

  transform: translateY(250px);
  z-index: 7;
  transition: all 400ms cubic-bezier(0.215, 0.61, 0.355, 1);
}

.search-content.active {
  transform: translateY(0);
  z-index: 9;
}

.search-content-btn-close {
  border: 0;
  background: none;
  position: absolute;
  top: 20px;
  right: 20px;
  font-size: 1.6em;
  cursor: pointer;
  color: #ccc;
  outline: none;
}

.search-content-btn-close:hover {
  opacity: 0.75;
}

.search-content-results {
  position: relative;
  height: 100%;
  border-spacing: 0;
}

.search-content-results thead,
.search-content-results tfoot,
.search-content-results tbody tr {
  display: table;
  width: 100%;
  table-layout: fixed;
}

.search-content-results thead {
  border-bottom: 4px solid rgba(0, 0, 0, 0.05);
}

.search-content-results thead th {
  border-bottom: 1px solid #333;
  height: 50px;
}

.search-content-results tfoot {
  background-color: white;
}

.search-content-results tfoot th {
  height: 60px;
}

.search-content-results tbody {
  position: absolute;
  left: 0;
  right: 0;
  top: 50px;
  bottom: 60px;
  padding: 10px 0;
  overflow-y: auto;
  overflow-x: hidden;
}

.search-content-results .row-tools-head,
.search-content-results .row-tools-body {
  width: 120px;
}

.search-content-results .type-head,
.search-content-results .type-body {
  width: 25%;
}

.search-content-results tr td {
  position: relative;
  cursor: pointer;
  font-size: 0.95em;
  color: #666;
}

.search-content-results tr:hover td {
  color: #2eb78e;
}

.search-content-results tr:hover button {
  color: #2eb78e;
  opacity: 0.5;
}

.search-content-results th,
.search-content-results td {
  text-align: left;
  color: #222;
  font-size: 1em;
  border-bottom: 1px solid #f2f2f2;
}

.search-content-results th {
  font-weight: bold;
  font-size: 1.2em;
}

.search-content-results tr:hover .row-tools {
  display: flex;
}

.search-content-results .row-tools button {
  padding: 8px 8px;
  font-size: 1.0em;
  background: none;
  margin-left: 0 10px;
  border: none;
  opacity: 0.05;
  width: 48px;
}

.search-content-results .row-tools button:hover {
  opacity: 1;
}

.search-content-base {
  position: absolute;
  bottom: 0;
  display: flex;
  align-items: center;
  height: 60px;
  border-top: 1px solid rgba(0, 0, 0, 0.3)
}
