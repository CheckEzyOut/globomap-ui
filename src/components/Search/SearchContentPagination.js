/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { findNodes } from '../../redux/modules/nodes';
import './SearchContentPagination.css';

export class SearchContentPagination extends Component {

  constructor(props) {
    super(props);

    this.state = {
      pageNumber: 1
    };

    this.onSendSearchQuery = this.onSendSearchQuery.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.doFindNodes = this.doFindNodes.bind(this);
  }

  doFindNodes(pageNumber) {
    if (pageNumber <= 0 || pageNumber > this.props.totalPages) {
      return;
    }

    this.props.findNodes({
      ...this.props.searchOptions,
      page: pageNumber
    });
  }

  onSendSearchQuery(event, dir) {
    event.preventDefault();
    let page = this.state.pageNumber;

    if (dir === 'next') {
      page = page + 1;
    } else if (dir === 'previous') {
      page = page - 1;
    }

    this.doFindNodes(page);
  }

  handleInputChange(event) {
    let pageNumber = event.target.value.trim();

    if (pageNumber.length === 0) {
      this.setState({ pageNumber });
      return;
    }

    pageNumber = parseInt(event.target.value, 10) || 1;
    this.setState({ pageNumber });
  }

  handleKeyPress(event) {
    let pageNumber = 0;
    if (event.key !== 'Enter') {
      return;
    }

    event.preventDefault();
    pageNumber = parseInt(event.target.value, 10) || 1;
    this.doFindNodes(pageNumber);
  }

  componentDidUpdate(prevProps) {
    if(prevProps.currentPage !== this.props.currentPage) {
      this.setState({ pageNumber: this.props.currentPage });
    }
  }

  render() {
    if (this.props.nodeList === undefined || this.props.nodeList.length === 0) {
      return null;
    }

    return (
      <div className="search-content-pagination">
        <button className="btn-previous gmap-btn small" onClick={(e) => this.onSendSearchQuery(e, 'previous')}
          disabled={this.props.currentPage === 1}>
          <i className="fa fa-angle-left"></i>
        </button>

        <input className="page-number gmap-text small" type="text" name="pagination"
          value={this.state.pageNumber} onChange={this.handleInputChange} onKeyPress={this.handleKeyPress} />

        <button className="btn-next gmap-btn small" onClick={(e) => this.onSendSearchQuery(e, 'next')}
          disabled={this.props.currentPage === this.props.totalPages}>
          <i className="fa fa-angle-right"></i>
        </button>

        <div className="pagination-info">
          {this.props.totalPages} Page{this.props.totalPages > 1 && 's'}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { nodeList, totalPages,
          currentPage, searchOptions } = state.nodes;
  return {
    nodeList,
    totalPages,
    currentPage,
    searchOptions
  };
}

export default connect(
  mapStateToProps,
  { findNodes }
)(SearchContentPagination);
