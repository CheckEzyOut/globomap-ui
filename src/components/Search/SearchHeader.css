.search-header * {
  box-sizing: border-box;
}

.search-header {
  display: flex;
  width: 100%;
  flex-flow: column;
}

.search-header button {
  background: none;
  border: none;
  cursor: pointer;
  margin: 0 10px 10px 0;
  outline: none;
}

.search-header button:hover {
  opacity: 0.75;
}

.search-top {
  display: flex;
}

.search-box .btn-search {
  font-size: 2em;
  height: 42px;
}

.toggle-search-type {
  position: relative;
  width: 180px;
}

.search-box .prop-query,
.search-box .search-query {
  display: block;
  width: 100%;
  border: none;
  font-size: 8em;
  font-weight: 300;
  line-height: normal;
  color: #2eb78e;
  padding: 50px 10px 20px;
  border-bottom: 1px solid #eee;
  margin-bottom: 40px;
  outline: none;

  transition: all 600ms cubic-bezier(0.19, 1, 0.22, 1);
}

.with-options .search-box .search-query {
  font-size: 3em;
  padding: 30px 10px 10px;
}

.search-box .prop-query::placeholder,
.search-box .search-query::placeholder {
  color: #eee;
}

.search-for-properties {
  margin-bottom: 20px;
}

.search-for-properties .btn-add-prop {
  font-size: 0.8em;
}

.search-for-properties .btn-add-prop .icon {
  color: #2eb78e;
}

.property-items {
  margin-bottom: 20px;
  max-height: 300px;
  overflow-y: auto;
}

.property-items .prop-item {
  display: flex;
  justify-content: left;
  flex-wrap: nowrap;
}

.prop-item .prop-operator {
  border: none;
  background: none;
  color: #2eb78e;
  outline: none;
  margin: 0 20px;
  font-size: 2em;
}

.prop-item .btn-remove-prop {
  font-size: 2em;
  color: #ccc;
  margin-left: 20px;
}

.prop-item .prop-query {
  font-size: 2em;
  width: 30%;
  padding: 30px 10px 10px;
  margin-bottom: 20px;
}

.search-types,
.selected-search-type {
  position: absolute;
  top: 0;
  left: 0;
  color: #ccc;
  width: 100%;
  border: 1px solid #ccc;
  border-radius: 21px;
  padding: 12px 20px;
  height: 42px;
  cursor: pointer;
  background-color: white;
  z-index: 9;
}

.search-types {
  height: 125px;
  padding: 56px 20px 0 38px;
  z-index: 8;
}

.search-types button:disabled {
  color: #2eb78e;
}

.selected-search-type .arrow {
  float: right;
}

.selected-search-type .search-type {
  color: #2eb78e;
}

.btn-search-options {
  float: right;
}

.search-options {
  overflow-y: auto;
  border-top: 2px solid #ccc;
  padding: 10px 0 0;
}

.search-options .collection-items {
  display: flex;
  flex-flow: row;
  flex-wrap: wrap;
  justify-content: space-between;
}

.search-options .collection-items .item {
  width: 30%;
  min-width: 200px;
  border-bottom: 1px solid #ccc;
  margin: 8px 5px;
  padding: 8px;
  cursor: pointer;
}

.search-options .collection-items .item:hover {
  border-color: #2eb78e;
}

.search-options .collection-items .item.checked {
  color: #2eb78e;
}

.search-options .collection-items .item input {
  float: right;
}
