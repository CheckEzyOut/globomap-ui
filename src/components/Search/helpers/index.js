export const collections = [
  {
    "alias": "Unknown",
    "name": "unknown",
    "kind": "document",
    "icon": "unknown",
    "description": "Unknown",
    "users": [
      "u_globomap_driver_acs",
      "u_globomap_driver_cmdb",
      "u_globomap_driver_napi"
    ]
  }
]
