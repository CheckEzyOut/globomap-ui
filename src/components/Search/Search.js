/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  SearchHeader,
  SearchContent } from '../';
import './Search.css';

export class Search extends Component {

  render() {
    return (
      <div className={`search-container ${this.props.className}`}>
        <section className="search base-content">
          <div className="base-content-header">
            <h2 className="base-content-title">Advanced Search</h2>
          </div>
          <SearchHeader />
        </section>
        <SearchContent />
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {};
}

export default connect(
  mapStateToProps,
  {}
)(Search);
