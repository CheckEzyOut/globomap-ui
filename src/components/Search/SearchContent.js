/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  clearNodes,
  setCurrentNode,
  clearCurrentNode
} from '../../redux/modules/nodes';
import { addStageNode } from '../../redux/modules/stage';
import { setTab } from '../../redux/modules/tabs';
import SearchContentPagination from './SearchContentPagination';
import { Properties } from '../';
import './SearchContent.css';

class SearchContent extends Component {

  constructor(props) {
    super(props);
    this.onNodeSelect = this.onNodeSelect.bind(this);
  }

  onNodeSelect(event, node) {
    event.stopPropagation();
    if (this.props.currentNode !== null &&
        this.props.currentNode._id === node._id) {
      return this.props.clearCurrentNode();
    }
    return this.props.setCurrentNode(node);
  }

  onOpenMap(event, node) {
    event.stopPropagation();
    this.props.addStageNode(node, null, true);
    // change to map screen
    this.props.setTab('map');
  }

  onOpenProperties(event, node) {
    event.stopPropagation();
  }

  onCloseSearchContent(event) {
    event.stopPropagation();
    this.props.clearNodes();
    this.props.clearCurrentNode();
  }

  render() {
    let index = this.props.perPage * (this.props.currentPage - 1);
    index = (index === 0 ? 1 : index + 1);

    let allNodes = [];
    if (this.props.nodeList !== undefined) {
      allNodes = this.props.nodeList.map((node, i) => {
        let current = (this.props.currentNode &&
                       node._id === this.props.currentNode._id);
        return (
          <React.Fragment key={node._id}>
            <tr className={current ? 'current' : ''}
                onClick={e => this.onNodeSelect(e, node)}>
              <td>{i + index}</td>
              <td>{this.props.namedCollections[node.type].alias}</td>
              <td>{node.name}</td>
            </tr>
            <tr className={`node-options${current ? ' open' : ''}`}>
              <td colSpan="3">
                <button onClick={e => this.onOpenProperties(e, node)}>
                  <i className="icon fa fa-info" /> Properties
                </button>
                <button onClick={e => this.onOpenMap(e, node)}>
                  <i className="icon fa fa-sitemap" /> View Map
                </button>
                {/* <div className="node-properties">
                  <Properties item={node} />
                </div> */}
              </td>
            </tr>
          </React.Fragment>
        );
      });
    }

    return (
      <div className={`search-content${allNodes.length > 0 ? ' active' : ''}`}>
        <button className="search-content-btn-close"
                onClick={e => this.onCloseSearchContent(e)}>
          <i className="fa fa-times"></i>
        </button>
        {allNodes.length > 0 &&
          <table className="search-content-results">
            <thead>
              <tr>
                <th>#</th>
                <th>Type</th>
                <th>Name</th>
              </tr>
            </thead>
            <tbody>{allNodes}</tbody>
            <tfoot className="search-content-base">
              <tr>
                <th><SearchContentPagination /></th>
              </tr>
            </tfoot>
          </table>}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    nodeList: state.nodes.nodeList,
    currentNode: state.nodes.currentNode,
    perPage: state.nodes.perPage,
    currentPage: state.nodes.currentPage,
    namedCollections: state.app.namedCollections
  };
}

export default connect(
  mapStateToProps,
  { addStageNode, setCurrentNode, clearNodes,
    clearCurrentNode, setTab }
)(SearchContent);
