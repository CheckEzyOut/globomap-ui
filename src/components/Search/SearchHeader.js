/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  resetSubNodes,
  findNodes } from '../../redux/modules/nodes';
import { cleanStageNodes } from '../../redux/modules/stage';
import { sortBy } from '../../utils';
import './SearchHeader.css';

export class SearchHeader extends Component {

  constructor(props) {
    super(props);
    this.state = {
      checkedCollections: [],
      query: '',
      showOptions: false,
      showSearchTypes: false,
      searchFor: "name",
      queryProps: [{ 'name': '', 'value': '', 'op': 'LIKE' }],
      propsOperators: ["==", "LIKE", "NOTIN", "IN", "!=", ">", ">=", "<", "<=", "!~", "=~"]
    };

    this.handleCheckItem = this.handleCheckItem.bind(this);
    this.handleSearchChange = this.handleSearchChange.bind(this);
    this.onToggleSearchOptions = this.onToggleSearchOptions.bind(this);
    this.closeSearchOptions = this.closeSearchOptions.bind(this);
  }

  addProp(event) {
    event.preventDefault();
    this.propItems.scrollTop = this.propItems.scrollHeight;
    let qProps = this.state.queryProps.slice();
    qProps.push({ 'name': '', 'value': '', 'op': 'LIKE' });
    this.setState({ queryProps: qProps });
  }

  removeProp(event, index) {
    event.preventDefault();
    let qProps = this.state.queryProps.slice();
    qProps.splice(index, 1);
    this.setState({ queryProps: qProps });
  }

  handlePropChange(event, index, item) {
    let qProps = this.state.queryProps.slice();
    qProps[index][item] = event.target.value;
    this.setState({ queryProps: qProps });
  }

  handleCheckItem(event) {
    let target = event.target,
        co = this.state.checkedCollections.slice(),
        itemIndex = co.indexOf(target.name);

    if (itemIndex < 0) {
      co.push(target.name);
    } else {
      co.splice(itemIndex, 1);
    }

    this.setState({ checkedCollections: co });
  }

  handleSearchChange(event) {
    let target = event.target;
    let value = target.type === 'checkbox' ? target.checked : target.value;
    this.setState({ [target.name]: value });
  }

  handleKeyPress(event) {
    if (event.key === 'Enter') {
      event.preventDefault();
      this.onSendSearchQuery(event);
    }
  }

  onSendSearchQuery(event) {
    event.preventDefault();

    this.props.cleanStageNodes();
    this.props.resetSubNodes();

    let checked = this.state.checkedCollections;

    if (checked.length === 0) {
      checked = this.props.collections.map(co => co.name);
    }

    let params = { collections: checked };

    if (this.state.searchFor === 'all') {
      params['query'] = this.state.query;
    } else if (this.state.searchFor === 'name') {
      params['query'] = this.state.query;
      params['queryType'] = 'name';
    } else {
      params['queryProps'] = this.state.queryProps;
    }

    this.props.findNodes(params);
  }

  onToggleSearchOptions(event) {
    event.preventDefault();
    this.setState({ showOptions: !this.state.showOptions });
  }

  closeSearchOptions() {
    this.setState({ showOptions: false });
  }

  render() {
    const coItems = sortBy(this.props.collections, 'alias');
    const collectionItems = coItems.map((co) => {
      const checked = this.state.checkedCollections.includes(co.name);
      return (
        <label key={co.name} className={`item${checked ? ' checked' : ''}`}>
          {co.alias}
          <input type="checkbox" name={co.name} checked={checked} onChange={this.handleCheckItem} />
        </label>
      );
    });

    const propItems = this.state.queryProps.map((prop, i) => {
      return (
        <div key={'prop' + i} className="prop-item">
          <input className="prop-query" type="search"
            name={'prop'+i+'-name'} placeholder="property name" value={this.state.queryProps[i].name}
            onChange={e => this.handlePropChange(e, i, 'name')} />

          <select className="prop-operator" name={'prop'+i+'-operator'}
            value={this.state.queryProps[i].op} onChange={e => this.handlePropChange(e, i, 'op')}>
            {this.state.propsOperators.map((op, j) => <option key={'op'+j} value={op}>{op}</option>)}
          </select>

          <input className="prop-query" type="search"
            name={'prop'+i+'-value'} placeholder="value" value={this.state.queryProps[i].value}
            onChange={e => this.handlePropChange(e, i, 'value')}
            onKeyPress={e => this.handleKeyPress(e)} />

          {i > 0 &&
            <button className="btn-remove-prop" onClick={e => this.removeProp(e, i)}>
              <i className="fas fa-times"></i>
            </button>}
        </div>
      );
    });

    return (
      <header className={`search-header${this.state.showOptions ? ' with-options' : ''}`}>

        <div className="search-box">
          <div className="search-top">
            <button className="btn btn-search" onClick={e => this.onSendSearchQuery(e)}
              disabled={this.props.findLoading}>
                <i className="fa fa-search"></i> Search
            </button>

            <div className="toggle-search-type">
              <div className="selected-search-type"
                  onClick={() => this.setState({ showSearchTypes: !this.state.showSearchTypes })}>
                for <span className="search-type">{this.state.searchFor}</span> <i className="arrow fas fa-chevron-down"></i>
              </div>
              {this.state.showSearchTypes &&
                <div className="search-types">
                  <button onClick={() => this.setState({ searchFor: 'all', showSearchTypes: false })}
                    disabled={this.state.searchFor === 'all'}>all</button>

                  <button onClick={() => this.setState({ searchFor: 'name', showSearchTypes: false })}
                    disabled={this.state.searchFor === 'name'}>name</button>

                  <button onClick={() => this.setState({ searchFor: 'properties', showSearchTypes: false })}
                    disabled={this.state.searchFor === 'properties'}>properties</button>
                </div>}
            </div>
          </div>

          {this.state.searchFor === 'all' &&
            <div className="search-for-all">
              <input className="search-query" type="search" name="query"
                placeholder="Type keyword" autoComplete="off"
                value={this.state.query}
                onChange={this.handleSearchChange}
                onKeyPress={e => this.handleKeyPress(e)} />
            </div>}

          {this.state.searchFor === 'name' &&
            <div className="search-for-all">
              <input className="search-query" type="search" name="query"
                placeholder="Type keyword" autoComplete="off"
                value={this.state.query}
                onChange={this.handleSearchChange}
                onKeyPress={e => this.handleKeyPress(e)} />
            </div>}

          {this.state.searchFor === 'properties' &&
            <div className="search-for-properties">
              <div className="property-items" ref={(el) => { this.propItems = el; }}>
                {propItems}
              </div>
              <button className="btn-add-prop" onClick={e => this.addProp(e)}>
                <i className="icon fas fa-plus"></i> Add another search
              </button>
            </div>}

          <button className="btn btn-search-options" onClick={(e) => this.onToggleSearchOptions(e)}>
            <i className="fa fa-sliders-h"></i> Toggle filters
          </button>
        </div>

        {this.state.showOptions &&
          <div className="search-options">
            <div className="collection-items">
              {collectionItems}
            </div>
          </div>}
      </header>
    );
  }

}

function mapStateToProps(state) {
  return {
    graphs: state.app.graphs,
    collections: state.app.collections,
    collectionsByGraphs: state.app.collectionsByGraphs,
    selectedCollections: state.app.selectedCollections,
    findLoading: state.nodes.findLoading
  };
}

export default connect(
  mapStateToProps,
  {
    resetSubNodes,
    findNodes,
    cleanStageNodes
  }
)(SearchHeader);
