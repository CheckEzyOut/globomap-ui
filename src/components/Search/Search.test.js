import React from 'react';
import { shallow } from 'enzyme';
import { Search } from './Search';

describe("Search", function() {

  it('should render Search component', () => {
    const component = shallow(<Search />);
    expect(component).toMatchSnapshot();
  });

});
