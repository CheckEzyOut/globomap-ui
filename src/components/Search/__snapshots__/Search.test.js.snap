// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render Search component 1`] = `
<div
  className="search-container"
>
  <section
    className="search"
  >
    <Connect(SearchHeader) />
  </section>
  <Connect(SearchContent) />
</div>
`;
